File: app/src/main/java/wannabit/io/cosmostaion/base/chains/Coreum.java
package wannabit.io.cosmostaion.base.chains;

import static wannabit.io.cosmostaion.base.BaseConstant.COINGECKO_URL;
import static wannabit.io.cosmostaion.base.BaseConstant.EXPLORER_BASE_URL;

import wannabit.io.cosmostaion.R;
import wannabit.io.cosmostaion.base.BaseChain;

public class Coreum extends ChainConfig {

    public BaseChain baseChain() { return BaseChain.COREUM_MAIN; }
    public int chainImg() { return R.drawable.chain_coreum; }
    public int chainInfoImg() { return R.drawable.infoicon_coreum; }
    public int chainInfoTitle() { return R.string.str_front_guide_title_coreum; }
    public int chainInfoMsg() { return R.string.str_front_guide_msg_coreum; }
    public int chainColor() { return R.color.color_coreum; }
    public int chainBgColor() { return R.color.colorTransBgCoreum; }
    public int chainTabColor() { return R.color.color_tab_myvalidator_coreum; }
    public String chainName() { return "coreum"; }
    public String chainKoreanName() { return "코리움"; }
    public String chainIdPrefix() { return "testcore-"; }

    public int mainDenomImg() { return R.drawable.token_coreum; }
    public String mainDenom() { return "utestcore"; }
    public int decimal() {
        return 9;
    }
    public String addressPrefix() { return "testcore"; }

    public boolean dexSupport() { return false; }
    public boolean wcSupport() { return false; }
    public boolean authzSupport() { return false; }

    public String grpcUrl() { return "grpc-coreum.cosmostation.io"; }

    public String explorerUrl() { return EXPLORER_BASE_URL + "coreum/"; }
    public String homeInfoLink() { return  "https://www.coreum.com/"; }
    public String blogInfoLink() { return  ""; }
    public String coingeckoLink() { return  COINGECKO_URL + "coreum"; }
}

File: app/src/main/java/wannabit/io/cosmostaion/base/chains/Coreum.kt
package wannabit.io.cosmostaion.base.chains

import com.google.common.collect.ImmutableList
import org.bitcoinj.crypto.ChildNumber
import wannabit.io.cosmostaion.R
import wannabit.io.cosmostaion.base.BaseChain
import wannabit.io.cosmostaion.base.BaseConstant
import wannabit.io.cosmostaion.base.BaseConstant.COINGECKO_URL

class Coreum : ChainConfig() {

    override fun baseChain(): BaseChain { return BaseChain.COREUM_MAIN }
    override fun chainImg(): Int { return R.drawable.chain_coreum }
    override fun chainInfoImg(): Int { return R.drawable.infoicon_coreum }
    override fun chainInfoTitle(): Int { return R.string.str_front_guide_title_coreum}
    override fun chainInfoMsg(): Int { return R.string.str_front_guide_msg_coreum }
    override fun chainColor(): Int { return R.color.color_coreum }
    override fun chainBgColor(): Int { return R.color.colorTransBgCoreum }
    override fun chainTabColor(): Int { return R.color.color_tab_myvalidator_coreum }
    override fun chainName(): String { return "coreum" }
    override fun chainKoreanName(): String { return "코리움" }
    override fun chainTitle(): String { return "coreum" }
    override fun chainIdPrefix(): String { return "coreum-mainnet-" }

    override fun mainDenomImg(): Int { return R.drawable.token_coreum }
    override fun mainDenom(): String { return "ucore" }
    override fun addressPrefix(): String { return "core" }

    override fun dexSupport(): Boolean { return false }
    override fun wcSupport(): Boolean { return false }
    override fun authzSupport(): Boolean { return true }

    override fun grpcUrl(): String { return "grpc-coreum.cosmostation.io" }
    override fun explorerUrl(): String { return BaseConstant.EXPLORER_BASE_URL + "coreum/" }
    override fun homeInfoLink(): String { return "https://www.coreum.com/" }
    override fun blogInfoLink(): String { return "https://www.coreum.com/community#press-&-media" }
    override fun coingeckoLink(): String { return COINGECKO_URL + "coreum" }

    override fun defaultPath(): String { return "m/44'/990'/0'/0/X" }

    override fun setParentPath(customPath: Int): List<ChildNumber>? {
        return ImmutableList.of(ChildNumber(44, true), ChildNumber(990, true), ChildNumber.ZERO_HARDENED, ChildNumber.ZERO)
    }
}